zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/FormResourceTransformer.java

```diff
@@ -91,15 +92,14 @@ public Either<Failure, Void> writeRecords(
     return Either.right(null);
   }
 
-  private Either<Failure, String> parseFormId(final DeploymentResource resource) {
+  private Either<Failure, Form> parseForm(final DeploymentResource resource) {
     try {
-      final String formId = JSON_MAPPER.readValue(resource.getResource(), FormIdPOJO.class).getId();
-
-      return validateFormId(formId);
+      final var form = JSON_MAPPER.readValue(resource.getResource(), Form.class);
+      return validateForm(form);
     } catch (final JsonProcessingException e) {
       final var failureMessage =
           String.format(
-              "Failed to parse formId from form JSON. '%s': %s",
+              "Failed to parse form JSON. '%s': %s",
```

Comment:
🤔 I think this is a potential breaking change, as some users may parse these messages and react on them.

@korthout do you think changing this message is acceptable? Forms are a "relatively" new feature. Furthermore, we don't guarantee the content of error messages (afaik).

Reply:
@koevskinikola I get this concern, but so far we've never made an issue out of this AFAIK. If we were to take our promises about not breaking user space to this level, it would become very hard to further improve things. I.e. in order to change error messages we'd need some configuration option to use the "improved" error messages.

I'd argue that:
- the error type can be depended upon as they are documented in the gRPC API (and perhaps also in the REST API 🤔)
- the error message provide additional information, but are not documented and can thus not be depended upon

Reply:
Thanks! So nothing to be changed here :).